src/Parser.ts

```diff
@@ -115,12 +115,22 @@ export class Parser extends EventEmitter {
   }
 
   protected isMetadataChanged (metadata: Map<string, string>): boolean {
-    for (const [key, value] of metadata.entries()) {
-      if (this.previousMetadata.get(key) !== value) {
-        return true;
+    if (this.options.notifyOnChangeOnly.length > 0) {
```

Comment:
Now, here we can check the type of the option. If there is an array, we treat it by your logic but if there is a boolean, nothing changes and works as before.

Reply:
```ts
if (Array.isArray(this.options.notifyOnChangeOnly)) {
  // your new logic
} else if (this.options.notifyOnChangeOnly === true) {
  // old logic
}
```